sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MonitorQueryTimeInterval.java

```diff
@@ -0,0 +1,239 @@
+     * @throws IllegalArgumentException if {@code value} is not in the correct format.
+     */
+    public static MonitorQueryTimeInterval parse(String value) {
+        Objects.requireNonNull(value);
+
+        String[] parts = value.split("/");
+        if (parts.length == 1) {
+            // duration
+            Duration duration = parseDuration(parts[0]);
+            if (duration == null || parts[0].length() + 1 == value.length()) {
+                // input strings like "PT24H/" are invalid
+                throw LOGGER.logExceptionAsError(
+                        new IllegalArgumentException(String.format(ERROR_MESSAGE, value)));
+            }
+
+            return new MonitorQueryTimeInterval(duration);
+        }
+
+        if (parts.length == 2) {
+            Duration startDuration = parseDuration(parts[0]);
+            OffsetDateTime startTime = parseTime(parts[0]);
+
+            Duration endDuration = parseDuration(parts[1]);
+            OffsetDateTime endTime = parseTime(parts[1]);
+
+            if (startDuration != null && endTime != null) {
+                return new MonitorQueryTimeInterval(startDuration, endTime);
+            }
+            if (startTime != null && endTime != null) {
+                return new MonitorQueryTimeInterval(startTime, endTime);
+            }
+            if (startTime != null && endDuration != null) {
+                return new MonitorQueryTimeInterval(startTime, endDuration);
+            }
+        }
+        throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException(String.format(ERROR_MESSAGE, value)));
+    }
+
+    @Override
+    public String toString() {
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        if (duration != null && endTime != null) {
+            sb.append("duration: ")
+                    .append(duration)
+                    .append(", ")
+                    .append("end time: ")
+                    .append(endTime);
+        } else if (startTime != null && endTime != null) {
+            sb.append("start time: ")
+                    .append(startTime)
+                    .append(", ")
+                    .append("end time: ")
+                    .append(endTime);
+        } else if (startTime != null && duration != null) {
+            sb.append("start time: ")
+                    .append(startTime)
+                    .append(", ")
+                    .append("duration: ")
+                    .append(duration);
+        } else {
+            sb.append("duration: ").append(duration);
+        }
+        sb.append("]");
+        return sb.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MonitorQueryTimeInterval that = (MonitorQueryTimeInterval) o;
+
+        return Objects.equals(this.duration, that.duration)
```

Comment:
Would this not rely on equals override on individual types like duration etc. Do we want to rely on that or explicitly set equality to be value equals?

Reply:
Ugh - I think I misunderstood this one.